public/app/plugins/panel/dashlist/DashList.tsx

```diff
@@ -0,0 +1,142 @@
+import { PanelProps } from '@grafana/data';
+import React, { useEffect, useState } from 'react';
+import { DashListOptions } from './types';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+import { DashboardSearchHit } from 'app/features/search/types';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+import { CustomScrollbar, Icon } from '@grafana/ui';
+import impressionSrv from 'app/core/services/impression_srv';
+import take from 'lodash/take';
+
+type Dashboard = DashboardSearchHit & { isSearchResult?: boolean; isRecent?: boolean };
+
+interface DashboardGroup {
+  show: boolean;
+  header: string;
+  dashboards: Dashboard[];
+}
+
+export function DashList(props: PanelProps<DashListOptions>) {
+  const [dashboards, setDashboards] = useState<Record<number, Dashboard>>({});
+  useEffect(() => {
+    async function fetchDashboards() {
```

Comment:
@kaydelaney I'm not a big fan of having the function implementation in a component this way. I suggest extracting this function outside of the component (making it accept respective arguments) and instead of using the `useEffect` you could use https://github.com/streamich/react-use/blob/master/docs/useAsync.md instead

Reply:
I suppose this could be a `PureComponent` as well?

Reply:
Then we won't be able to use hooks.